src/files.ts

```diff
@@ -76,24 +74,38 @@ export function parseFileIgnores(path: string): string[] {
     }
   }
 
-  return rules.map(rule => {
-    let prefix = "";
+  return rules.reduce((results: string[], rule) => {
+    let prefix = '';
     if (rule.startsWith('!')) {
+      // eslint-disable-next-line no-param-reassign
       rule = rule.substring(1);
-      prefix = "!";
+      prefix = '!';
     }
-    return rule.startsWith('/') || rule.startsWith('**')
-      ? prefix + nodePath.posix.join(dirname, rule, '**')
-      : prefix + nodePath.posix.join(dirname, '**', rule, '**');
-  });
+    // Mappings from .gitignore format to glob format:
```

Comment:
optional: I have spent quite some time to try and understand what's going on. I think the mapping would benefit from being split into a separate function.